pkg/reconciler/pipelinerun/cancel.go

```diff
@@ -59,6 +60,39 @@ func init() {
 
 // cancelPipelineRun marks the PipelineRun as cancelled and any resolved TaskRun(s) too.
 func cancelPipelineRun(ctx context.Context, logger *zap.SugaredLogger, pr *v1beta1.PipelineRun, clientSet clientset.Interface) error {
+	errs := cancelPipelineTaskRuns(ctx, logger, pr, clientSet)
+
+	// If we successfully cancelled all the TaskRuns and Runs, we can consider the PipelineRun cancelled.
+	if len(errs) == 0 {
+		reason := ReasonCancelled
+		if pr.Spec.Status == v1beta1.PipelineRunSpecStatusCancelledDeprecated {
+			reason = ReasonCancelledDeprecated
+		}
```

Comment:
thanks @rafalbigaj for adding this check 🙏 and setting it to deprecated reason when the deprecated status is set, it will be easier to cleanup and remove it from the project while upgarding to V1.